components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/wrappers/CibaAuthRequestWrapper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.ciba.wrappers;
+
+import org.wso2.carbon.identity.application.authentication.framework.model.CommonAuthRequestWrapper;
+import org.wso2.carbon.identity.oauth.ciba.common.CibaConstants;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Wrap ciba authentication request to remove unwanted params and added required params.
+ */
+public class CibaAuthRequestWrapper extends CommonAuthRequestWrapper {
+
+    // Map to accumulate additional parameters.
+    private Map<String, String> extraParameters;
+
+    public CibaAuthRequestWrapper(HttpServletRequest request) {
+
+        super(request);
+        extraParameters = new HashMap();
+    }
+
+    @Override
+    public String getParameter(String name) {
+
+        if (extraParameters.containsKey(name)) {
+            return extraParameters.get(name);
+        } else {
+            if ((CibaConstants.REQUEST.equals(name))) {
+                // In Authorize EndPoint we have defined (request) parameter already and hence to prevent conflict.
```

Comment:
Can we mention that we are removing 'request' parameter which denotes the CIBA request so that it does not conflict with 'request' param in OAuth2 which is usually used to denote a request object.